docs/settings.rst

```diff
@@ -39,193 +39,273 @@ Here is a list of settings for Pelican:
-===============================================================================  =====================================================================
+.. data:: USE_FOLDER_AS_CATEGORY = True
 
-.. [#] Default is the system locale.
+   When you don't specify a category in your post metadata, set this setting to
+   ``True``, and organize your articles in subfolders, the subfolder will
+   become the category of your post. If set to ``False``, ``DEFAULT_CATEGORY``
+   will be used as a fallback.
+
+.. data:: DEFAULT_CATEGORY = 'misc'
+
+   The default category to fall back on.
+
+.. data:: DISPLAY_PAGES_ON_MENU = True
+
+   Whether to display pages on the menu of the template. Templates may or may
+   not honor this setting.
+
+.. data:: DISPLAY_CATEGORIES_ON_MENU = True
+
+   Whether to display categories on the menu of the template. Templates may or
+   not honor this setting.
+
+.. data:: DOCUTILS_SETTINGS = {}
+
+   Extra configuration settings for the docutils publisher (applicable only to
+   reStructuredText). See `Docutils Configuration`_ settings for more details.
+
+.. data:: DELETE_OUTPUT_DIRECTORY = False
+
+   Delete the output directory, and **all** of its contents, before generating
+   new files. This can be useful in preventing older, unnecessary files from
+   persisting in your output. However, **this is a destructive setting and
+   should be handled with extreme care.**
+
+.. data:: OUTPUT_RETENTION = []
+
+   A list of filenames that should be retained and not deleted from the output
+   directory. One use case would be the preservation of version control data.
+
+   Example::
+
+      OUTPUT_RETENTION = [".hg", ".git", ".bzr"]
+
+.. data:: JINJA_EXTENSIONS = []
+
+   A list of any Jinja2 extensions you want to use.
+
+.. data:: JINJA_FILTERS = {}
+
+   A dictionary of custom Jinja2 filters you want to use.  The dictionary
+   should map the filtername to the filter function.
+
+   Example::
+
+    JINJA_FILTERS = {'urlencode': urlencode_filter}
+
+   See `Jinja custom filters documentation`_.
+
+.. data:: LOG_FILTER = []
+
+   A list of tuples containing the logging level (up to ``warning``) and the
+   message to be ignored.
+
+   Example::
+
+      LOG_FILTER = [(logging.WARN, 'TAG_SAVE_AS is set to False')]
+
+.. data:: READERS = {}
+
+   A dictionary of file extensions / Reader classes for Pelican to process or
+   ignore.
+
+   For example, to avoid processing .html files, set::
+
+      READERS = {'html': None}
+
+   To add a custom reader for the ``foo`` extension, set::
+
+      READERS = {'foo': FooReader}
+
+.. data:: IGNORE_FILES = ['.#*']
+
+   A list of glob patterns.  Files and directories matching any of these
+   patterns will be ignored by the processor. For example, the default
+   ``['.#*']`` will ignore emacs lock files, and ``['__pycache__']`` would
+   ignore Python 3's bytecode caches.
+
+.. data:: MD_EXTENSIONS = {...}
```

Comment:
@omiday: Did you rebase on current master? I only ask because I believe this setting — `MD_EXTENSIONS` — was recently replaced by the `MARKDOWN` setting via: https://github.com/getpelican/pelican/commit/a07c0e6e042a0b7a26a7f97ebd5e5eb977eea160#diff-f47f8ca652dac550c8bc9d449ca0d253L117